README.md

```diff
@@ -58,78 +145,173 @@ aws_secret_access_key=...
 aws_session_token=...
 ```
 
-## Get KMS key and CSR
+### Using LocalStack
+
+Instead of calling AWS services, you can use [LocalStack](https://www.localstack.cloud/) to run the example entirely on your local machine, simulating interactions with AWS. 
+
+NOTE: This setup is suitable for development only.
+
+#### Set up the LocalStack environment
+
+Install LocalStack following the [installation instructions](https://docs.localstack.cloud/getting-started/installation/) for your configuration.
+
+#### Run LocalStack
+
+Once LocalStack is installed, open a shell window and start the LocalStack stack in detached mode:
+
+```shell
+localstack start -d
+```
+
+Make sure to keep LocalStack running while you work through this example.
+Warning: Anything configured in LocalStack is transient, and will be lost on restart/reboot of the LocalStack container.
+
+#### Install awslocal CLI
+
+To interact with LocalStack, install the CLI tool `awslocal` that substitutes for the `aws` CLI while LocalStack is running. For more information, see the [LocalStack documentation](https://docs.localstack.cloud/user-guide/integrations/aws-cli/).
+
+Follow these steps:
+
+1. Ensure that you have activated your Python virtual environment.
+2. Since `awslocal` is a wrapper around the `aws` CLI, you need to install the `awscli` package first by entering this command:
+   ```shell
+    pip install awscli
+    ```
+3. Install `awslocal` into your local virtual environment by entering this command:
+    ```shell
+    pip install awscli-local
+    ```
+
+#### Create an environment file
+
+The app uses environment variables to get credentials and the LocalStack endpoint to access. Therefore, you need to set up a `.env` file in the root of the repository following the format of the [example-env.env file](example-env.env) file. The setup script and the Flask app will then get the values from the `.env` file.
```

Comment:
This explanation is ordered a bit backward.  Suggest:

The app gets the AWS credentials and the LocalStack endpoint to access from environment variables, which are set by the `.env` file in the root of the repository. To create this file:

1. Copy the provided template file [example-env.env file](https://github.com/contentauth/c2pa-python-example/blob/main/example-env.env):
```
cp example-env.env .env
```
2. Edit `.env` as needed to set the values described below.

Reply:
Done